server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodVersion.java

```diff
@@ -0,0 +1,90 @@
+package org.infinispan.server.hotrod;
+
+import java.util.Arrays;
+
+/**
+ * The various Hot Rod versions
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public enum HotRodVersion {
+   UNKNOWN(0, 0),
+   HOTROD_10(1, 0),
+   HOTROD_11(1, 1),
+   HOTROD_12(1, 2),
+   HOTROD_13(1, 3),
+   HOTROD_20(2, 0),
+   HOTROD_21(2, 1),
+   HOTROD_22(2, 2),
+   HOTROD_23(2, 3),
+   HOTROD_24(2, 4),
+   HOTROD_25(2, 5),
+   HOTROD_26(2, 6),
+   HOTROD_27(2, 7),
+   ;
+
+   private final int major;
+   private final int minor;
+   private final byte version;
+   private final String text;
+
+   HotRodVersion(int major, int minor) {
+      this.major = major;
+      this.minor = minor;
+      this.version = (byte) (major * 10 + minor);
+      this.text = version > 0 ? String.format("HOTROD/%d.%d", major, minor) : "UNKNOWN";
+   }
+
+   public byte getVersion() {
+      return version;
+   }
+
+   public boolean isNewerThan(byte version) {
+      return this.version > version;
+   }
+
+   public boolean isAtLeast(byte version) {
```

Comment:
This and the `isNewerThan` methods are throwing me off. From the names I would think that both are testing this `this` object is greater than the provided version (with the latter being true when equals as well). Maybe just add some documentation to the methods to clear it up.

Maybe `isOlderOrSameThan` instead here?

Reply:
yes, I'm not a fan of the "backwards" comparison: initially I had it the other way round, converting the byte version to a HotRodVersion and comparing that to a well known version, but then that would have lost all inlining because it would become a megamorphic call... I'll come up with some better names.